### ISSUE TITLE
Feature Request: Include Numerical Classification Metrics in Output (val=True)

### ISSUE THREAD

**Issue by VictorZani** [ultralytics/ultralytics#20515]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello Ultralytics team,

First, thank you for your outstanding work on the YOLOv8 framework. It has been instrumental in several computer vision projects I've undertaken.

I would like to propose an enhancement to the classification tasks within the framework. Currently, when performing classification with val=True, the output includes metrics like accuracy. However, it would be highly beneficial to also include the following numerical metrics:

Sensitivity (Recall)

Specificity

Receiver Operating Characteristic (ROC) Curve

Area Under the ROC Curve (AUC)

These metrics are standard in classification tasks and provide deeper insights into model performance, especially in scenarios with class imbalance or when the costs of false positives and false negatives differ.

Proposed Implementation:

  Calculate and display Sensitivity and Specificity for each class during validation.
  
  Generate and save ROC curves for each class, possibly as roc_curve_classname.png.
  
  Compute and report AUC values for each class, as well as macro and micro averages.
  
  Include these metrics in the validation logs and save them in a structured format (e.g., JSON) for further analysis.
  
Justification:

  Enhances the interpretability and evaluation of classification models.
  
  Aligns with standard practices in machine learning and statistical analysis.
  
  Provides users with comprehensive tools to assess model performance beyond accuracy alone.
  
  Additional Context:
  
While there have been discussions on computing these metrics manually using the confusion matrix and external libraries like scikit-learn, integrating them directly into the framework would streamline the evaluation process and reduce the overhead for users.

I am willing to contribute to the implementation or testing of this feature if needed.

Thank you for considering this enhancement.

Best regards,
Victor Hugo Zani

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @VictorZani, thank you for your thoughtful feature request and for your interest in Ultralytics 🚀! We appreciate your detailed suggestions on enhancing classification outputs in the framework.

This is an automated response to help you get the fastest possible support. An Ultralytics engineer will review your request and assist you further soon.

In the meantime, you might find it helpful to explore our [Docs](https://docs.ultralytics.com/) for more information on current classification metrics, [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and to see if any workarounds or related discussions are available.

If you have additional context, example outputs, or specific usage scenarios that would help clarify your needs, please feel free to share them below.

Join the Ultralytics community for real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧, in-depth discussion on [Discourse](https://community.ultralytics.com/), or to share insights on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

To make sure you’re testing with the latest features and improvements, please upgrade the `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can also test and develop in these verified environments (all dependencies preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your suggestion and willingness to contribute!
